apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/DocumentDetailActivity.java

```diff
@@ -165,6 +189,46 @@ public void onClick(View view) {
         return list;
     }
 
+    private void refreshDocumentWithRead(DocumentWrapper document){
+        if (document.getError() != null) {
+            String message = document.getError().getMessage();
```

Comment:
This block should be factored out if possible.  It also appears twice in DataActivity.java